src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,101 @@
-export const Pagination = () => {};
+import React, { useEffect } from 'react';
+import { getPartItems } from '../../utils';
+
+type Props = {
+  items: string[];
+  total: number;
+  perPage: number;
+  currentPage: number;
+  setElements: React.Dispatch<React.SetStateAction<string[]>>;
+  onPageChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const Pagination: React.FC<Props> = ({
+  items,
+  total,
+  perPage,
+  currentPage,
+  setElements,
+  onPageChange,
+}) => {
+  const amountOfButton = Math.ceil(total / Number(perPage));
```

Comment:
The variable name 'amountOfButton' is slightly misleading as it represents the amount of pages, not buttons. Consider renaming it to 'amountOfPages' for clarity.